Xamarin.Forms.Core/CornerRadius.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+
+namespace Xamarin.Forms.Core
+{
+	public struct CornerRadius
```

Comment:
As it's a struct, we need a way to figure out if the `CornerRadius` is the default CornerRadius, or if it was created by one of the parameterized `ctor`.

If we ever applied this property to, say, `Button`, having a default CornerRadius would mean "use whatever default on the platform".

There are multiple ways of implementing this, I'd recommend having a private field `isnotdefault` (we can not initialize fields with default values in structs) and set that field to `true` in the parameterized ctors.

That value should then be used in `Equals` and operators, so we can do `if (my box.CorenerRadius == default(CornerRadius)) {}`

Reply:
Added a IsDefault in CornerRadius class. Is it what is expected?

Reply:
here's what I meant

```csharp
	public struct CornerRadius
	{
		public double TopLeft { get; }		//as the properties are only set from the ctor, they do not need an internal setter
		public double TopRight { get; }
		public double BottomLeft { get; }
		public double BottomRight { get; }

		bool isParameterized;

		public CornerRadius(double uniformRadius) : this(uniformRadius, uniformRadius, uniformRadius, uniformRadius)
		{
		}

		public CornerRadius(double topLeft, double topRight, double bottomLeft, double bottomRight)
		{
			isParameterized = true;			//not default anymore
			TopLeft = topLeft;
			TopRight = topRight;
			BottomLeft = bottomLeft;
			BottomRight = bottomRight;
		}

		//...

		bool Equals(CornerRadius other)
		{
			if (!isParameterized && !other.isParameterized)
				return true; //default CornerRadius (created with parameterless ctor, or with default(CornerRadius) are Equal
			return TopLeft == other.TopLeft && TopRight == other.TopRight && BottomLeft == other.BottomLeft && BottomRight == other.BottomRight;
		}

		// ...
	}
```

Reply:
Ahh, now I understood you perfectly. Thank you!